Assets/Code/Scripts/PlaceBuilding.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlaceBuilding : MonoBehaviour
+{
+    public GameObject buildingPrefab;
+    public Transform playerTransform;
+    public KeyCode placeBuildingKey = KeyCode.T;
+
+    void Update()
+    {
+        // Sprawdź, czy gracz nacisnął przycisk "T"
+        if (Input.GetKeyDown(placeBuildingKey))
+        {
+            // Oblicz odległość przed graczem, w której ma zostać postawiony obiekt
+            float distance = 2.0f;
```

Comment:
Można by to wrzucić w SerializeField poza funkcją żeby z poziomu Unity dostosować do potrzeb